backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -403,102 +625,45 @@ func InitDBClient(initConnectionTimeout time.Duration) *storage.DB {
+	// This field is embedded via PipelineSpec in Job, and stored in the `jobs` table.
+	if err := db.Migrator().AlterColumn(&model.Job{}, "WorkflowSpecManifest"); err != nil {
+		glog.Fatalf("Failed to update WorkflowSpecManifest column to nullable. Error: %v", err)
 	}
 
-	switch driverName {
-	case "pgx":
-		response = db.Model(&model.RunMetric{}).
-			AddForeignKey("\"RunUUID\"", "run_details(\"UUID\")", "CASCADE" /* onDelete */, "CASCADE" /* onUpdate */)
-		if ignoreAlreadyExistError(driverName, response.Error) != nil {
-			glog.Fatalf("Failed to create a foreign key for RunUUID in run_metrics table. Error: %s", response.Error)
-		}
-		response = db.Model(&model.PipelineVersion{}).
-			AddForeignKey("\"PipelineId\"", "pipelines(\"UUID\")", "CASCADE" /* onDelete */, "CASCADE" /* onUpdate */)
-		if ignoreAlreadyExistError(driverName, response.Error) != nil {
-			glog.Fatalf("Failed to create a foreign key for PipelineId in pipeline_versions table. Error: %s", response.Error)
-		}
-		response = db.Model(&model.Task{}).
-			AddForeignKey("\"RunUUID\"", "run_details(\"UUID\")", "CASCADE" /* onDelete */, "CASCADE" /* onUpdate */)
-		if ignoreAlreadyExistError(driverName, response.Error) != nil {
-			glog.Fatalf("Failed to create a foreign key for RunUUID in task table. Error: %s", response.Error)
-		}
-	case "mysql":
-		response = db.Model(&model.RunMetric{}).
-			AddForeignKey("RunUUID", "run_details(UUID)", "CASCADE" /* onDelete */, "CASCADE" /* onUpdate */)
-		if ignoreAlreadyExistError(driverName, response.Error) != nil {
-			glog.Fatalf("Failed to create a foreign key for RunUUID in run_metrics table. Error: %s", response.Error)
-		}
-		response = db.Model(&model.PipelineVersion{}).
-			AddForeignKey("PipelineId", "pipelines(UUID)", "CASCADE" /* onDelete */, "CASCADE" /* onUpdate */)
-		if ignoreAlreadyExistError(driverName, response.Error) != nil {
-			glog.Fatalf("Failed to create a foreign key for PipelineId in pipeline_versions table. Error: %s", response.Error)
-		}
-		response = db.Model(&model.Task{}).
-			AddForeignKey("RunUUID", "run_details(UUID)", "CASCADE" /* onDelete */, "CASCADE" /* onUpdate */)
-		if ignoreAlreadyExistError(driverName, response.Error) != nil {
-			glog.Fatalf("Failed to create a foreign key for RunUUID in task table. Error: %s", response.Error)
-		}
-
-		// This is a workaround because AutoMigration does not detect that the column went from not null to nullable.
-		response = db.Model(&model.Job{}).ModifyColumn("WorkflowSpecManifest", client.MYSQL_TEXT_FORMAT_NULL)
-		if response.Error != nil {
-			glog.Fatalf("Failed to make the WorkflowSpecManifest column nullable on jobs. Error: %s", response.Error)
-		}
-	default:
-		glog.Fatalf("Driver %v is not supported, use \"mysql\" for MySQL, or \"pgx\" for PostgreSQL", driverName)
+	// Ensure the length modification in schema is enforced for legacy DBs
+	if err := ShrinkColumns(db, validation.LengthSpecs); err != nil {
```

Comment:
so to clarify, does AutoMigrate not handle column lengths, is that why we have to do this manually?